### PR TITLE
add CM4 support

### PR DESCRIPTION
**Pull request by mstormi**
Signed-off-by: Markus Storm <markus.storm@gmx.net>

### CODE REVIEW

File: functions/helpers.bash [hunk 1]
```diff
@@ -190,8 +190,17 @@ is_pifour_8GB() {
...
+  if [[ "$hw" == "cm4" ]]; then return 0; fi
+  grep -q "^Revision\\s*:\\s*[ 123][0-9a-fA-F][0-9a-fA-F][0-9a-fA-F]12[0-9a-fA-F]$" /proc/cpuinfo
+  return $?
+}
 is_pi() {
-  if is_pizero || is_pizerow || is_pione || is_cmone || is_pitwo || is_pithree || is_cmthree || is_pithreeplus || is_cmthreeplus || is_pifour; then return 0; fi
+  if is_pizero || is_pizerow || is_pione || is_cmone || is_pitwo || is_pithree || is_cmthree || is_pithreeplus || is_cmthreeplus || is_pifour || is_cmfour; then return 0; fi
+  return 1
+}
+is_cm() {}
```
Comment (ecdye) [inline functions/helpers.bash:202]: Error `is_cm() {` not `is_cm() {}`

**Review by ecdye — CHANGES_REQUESTED** [review]
Fix shellcheck errors

File: functions/helpers.bash (hunk 1 above)
Comment (mstormi) [reply functions/helpers.bash]: of course. Changed. Merge if you're ok now.

**Review by holgerfriedrich — COMMENTED** [review]
Where did you get the revision code `12` from? There is no update yet, 
https://www.raspberrypi.org/documentation/hardware/raspberrypi/revision-codes/README.md

**Review by ecdye — APPROVED** [review]

**Comment by mstormi** [comment]
> Where did you get the revision code `12` from?

Educated guessing on a number of internet sites. Granted that's a little premature but as that's unassigned yet it won't do harm either. I'll watch out for updates and change if needed.

**Comment by mstormi** [comment]
CM4 rev id is 14, changed

**Pull request merged**